plugins/template-validator.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const validator = require('screwdriver-template-validator');
+
+exports.register = (server, options, next) => {
+    server.route({
+        method: 'POST',
+        path: '/validator/template',
+        config: {
+            handler: (request, reply) => {
+                // rip eslint
+                const result = validator(request.payload.yaml)
+                .then((stuff) => {
+                    reply(stuff);
+                }, (validationErr) => {
+                    reply({
+                        error: validationErr
```

Comment:
We were using the `config-parser` as a guide for how to implement the `template-validator`. As you pointed out, this is one case where modeling the `template-validator` after the `config-parser` does not make sense. So we're looking for what is better for the caller.

After some thought, one alternative could be that this endpoint returns the following payload format: 

```
{
  data: {
    config: {
      // The template configuration object, or an empty object if it can't be parsed 
      ...
    },
    errors: [
      // The number of errors encountered while validating the template configuration
      // It would be empty if no errors were detected
      ...
    ]
  }
}
```

This payload has the benefit of giving the caller:

1. What the template looks like in Screwdriver. 
1. A list of errors that the validator found

Given these two things, the template owner has a minimum amount of information to determine why their template is wrong or is formatted differently than expected.